erpnext/stock/doctype/stock_entry_detail/stock_entry_detail.json

```diff
@@ -346,7 +346,8 @@
    "fieldtype": "Check",
    "label": "Allow Zero Valuation Rate",
    "no_copy": 1,
-   "print_hide": 1
+   "print_hide": 1,
+   "read_only_depends_on": "eval: doc.s_warehouse != \"\""
```

Comment:
`!doc.s_warehouse` would be better?

By default the field is `undefined` when user empties it it becomes `""`